src/hooks/useUploadImage.ts

```diff
@@ -0,0 +1,25 @@
+import {Image} from 'react-native-compressor';
+import {composeImageMeta} from '../utils/string/file';
+import {uploadPhoto} from '../service/file';
+
+const useUploadImage = () => {
+  const uploadPhotoImage = async (pathImg: string) => {
+    try {
+      const compressionResult = await Image.compress(pathImg, {
+        compressionMethod: 'auto'
+      });
+      const asset = new FormData();
+      asset.append('photo', composeImageMeta(compressionResult));
+      const responseUpload = await uploadPhoto(asset);
+      return responseUpload;
+    } catch (e) {
+      return undefined;
+    }
+  };
+
+  return {
+    uploadPhotoImage
+  };
+};
+
+export default useUploadImage;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useUploadImage` hook seems to be well implemented. However, there is a potential issue with error handling. In the catch block (lines 15-17), when an error occurs, it simply returns `undefined`. This could make debugging difficult if an error occurs during the image upload process because you won't know what the error was. It would be better to either log the error or throw it again after logging so that it can be handled upstream.

```diff
    } catch (e) {
-     return undefined;
+     console.error(e);
+     throw e;
    }
```
This way, you will have a record of the error and the calling code can decide how to handle it.


<!-- This is an auto-generated comment by OSS CodeRabbit -->